app/src/test/java/org/mozilla/focus/ext/UriKtTest.kt

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.ext
+
+import android.net.Uri
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class UriKtTest {
+
+    @Test
+    fun `WHEN truncatedHost is called on a null host THEN it returns the host`() {
+        val expectedHost = null
+        val uri = httpsUriBuilder()
+            .path("mozilla/en-US")
+            .authority(expectedHost)
+            .build()
+        assertNull(uri.host)
+
+        assertEquals(expectedHost, uri.truncatedHost())
+    }
+
+    @Test
+    fun `WHEN truncatedHost is called on an empty host THEN it returns the host`() {
+        val hosts = arrayOf(
+            "",
+            "    "
+        )
+        val uris = hosts.map {
+            httpsUriBuilder()
+                .path("mozilla/en-US")
+                .authority(it)
+                .build()
+        }
```

Comment:
Optional nit: it might be simpler to do the zipping in one step?  I'm honestly 50/50 on this one.

```kotlin
val uris = hosts.map { host ->
    val uri = httpsUriBuilder()
        .path("mozilla/en-US")
        .authority(host)
        .build()
    Pair(host, uri)
}
```

Reply:
I'm going to leave it as is – I agree, it could go either way so it's probably not worth changing.